analyses/molecular-subtyping-NBL/00-subset-for-NBL.Rmd

```diff
@@ -0,0 +1,146 @@
+
+#Load Libraries
+
+```{r load libraries}
+suppressPackageStartupMessages({
+library(dplyr)
+library(data.table)
+library(tidyverse)
+})
+```
+
+
+#Set up directories
+```{r setup directories}
+# root directory
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# data directory
+data_dir <- file.path(root_dir, "data")
+
+# module directory
+module_dir <- file.path(root_dir, "analyses","molecular-subtyping-NBL")
+
+#input directory
+input_dir <- file.path(root_dir, "analyses","molecular-subtyping-NBL","input")
+
+#results directory
+results_dir <- "results"
+# create if doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+
+#Load the files for histology, consensus, and tpm data
+```{r load the data files: this will take time}
+hist_df<-readr::read_tsv(file.path(data_dir, "histologies-base.tsv"),guess_max = 100000)
+
+consensus_df<-readr::read_tsv(file.path(data_dir, "consensus_wgs_plus_cnvkit_wxs.tsv.gz"),
+                              guess_max = 100000)
+
+gene_expression_df<-as.data.frame(readRDS(paste(data_dir,"/",
+                                                "gene-expression-rsem-tpm-collapsed.rds",sep="")))
+```
+
+Background: 
+NBL is defined as neuroblastoma, ganglioneuroblastoma, and ganglioneuroma.These diseases are 
+mentioned in the histology-base.tsv file. 
+
+consensus_wgs_plus_cnvkit_wxs.tsv.gz has information regarding copy number, status, and 
+gene symbol (MYCN). 
+
+gene-expression-rsem-tpm-collapsed.rds file has MYCN TPM values.
+
+
+Code Logic: 
+We first filter the histology-base.tsv file for NBL samples based on pathology_diagnosis, 
+sample_type,and experimental strategy. The pathology_diagnosis can have certain values as listed in
+the code chunk below. The sample type we query for is "tumor", i.e. we exclude "normal" samples. 
+For experimental strategy, we consider WGS, WXS, Targeted Sequencing, and RNA-Seq. We ultimately 
+want to create an output table with DNA and RNA biospecimen IDs, with their corresponding molecular 
+subtype, so we will need get DNA and RNA samples from the histology file. 
+
+We filter the consensus_wgs_plus_cnvkit_wxs.tsv.gz and gene-expression-rsem-tpm-collapsed.rds for
+MYCN to get the relevant information. 
+
+# Data Filtering
+```{r Data filtering}
+# First filter the histology file based on diseases we are intereseted in neuroblastoma, 
+# ganglioneuroblastoma, and ganglioneuroma
+
+hist_filtered_df <- hist_df %>% 
+  filter(sample_type=="Tumor", 
+         experimental_strategy %in% c("WGS", "WXS", "Targeted Sequencing", "RNA-Seq")) %>%
+  filter(pathology_diagnosis== "Neuroblastoma" |
+           pathology_diagnosis== "Ganglioneuroblastoma" |
+           pathology_diagnosis== "Ganglioneuroblastoma, nodular" |
+           pathology_diagnosis == "Ganglioneuroblastoma, intermixed" |
+           pathology_diagnosis == 
+           "Ganglioneuroma, maturing subtype OR Ganglioneuroblastoma, well differentiated")
```

Comment:
Hi @adilahiri - just a few suggestions to clean this up a bit. Can you rework this to create a JSON file with a terms list, similar to the other modules such as [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/molecular-subtyping-HGG/00-HGG-select-pathology-dx.R)?

Reply:
Thank you for the feedback @jharenza, will get this done.